Ryujinx.HLE/HOS/Services/Nv/INvDrvServices.cs

```diff
@@ -1,104 +1,300 @@
 using ARMeilleure.Memory;
+using Ryujinx.Common;
 using Ryujinx.Common.Logging;
+using Ryujinx.HLE.Exceptions;
 using Ryujinx.HLE.HOS.Ipc;
-using Ryujinx.HLE.HOS.Kernel.Common;
+using Ryujinx.HLE.HOS.Kernel.Memory;
 using Ryujinx.HLE.HOS.Kernel.Process;
-using Ryujinx.HLE.HOS.Kernel.Threading;
-using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvGpuAS;
-using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvGpuGpu;
+using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices;
+using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvHostAsGpu;
 using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvHostChannel;
 using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvHostCtrl;
+using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvHostCtrlGpu;
 using Ryujinx.HLE.HOS.Services.Nv.NvDrvServices.NvMap;
+using Ryujinx.HLE.HOS.Services.Nv.Types;
 using System;
 using System.Collections.Generic;
+using System.Reflection;
 
 namespace Ryujinx.HLE.HOS.Services.Nv
 {
     [Service("nvdrv")]
     [Service("nvdrv:a")]
+    [Service("nvdrv:s")]
+    [Service("nvdrv:t")]
     class INvDrvServices : IpcService
     {
-        private delegate int IoctlProcessor(ServiceCtx context, int cmd);
-
-        private static Dictionary<string, IoctlProcessor> _ioctlProcessors =
-                   new Dictionary<string, IoctlProcessor>()
-        {
-            { "/dev/nvhost-as-gpu",   ProcessIoctlNvGpuAS       },
-            { "/dev/nvhost-ctrl",     ProcessIoctlNvHostCtrl    },
-            { "/dev/nvhost-ctrl-gpu", ProcessIoctlNvGpuGpu      },
-            { "/dev/nvhost-gpu",      ProcessIoctlNvHostChannel },
-            { "/dev/nvhost-nvdec",    ProcessIoctlNvHostChannel },
-            { "/dev/nvhost-vic",      ProcessIoctlNvHostChannel },
-            { "/dev/nvmap",           ProcessIoctlNvMap         }
+        private static Dictionary<string, Type> _deviceFileRegistry =
+                   new Dictionary<string, Type>()
+        {
+                       { "/dev/nvmap",           typeof(NvMapDeviceFile)         },
+                       { "/dev/nvhost-ctrl",     typeof(NvHostCtrlDeviceFile)    },
+                       { "/dev/nvhost-ctrl-gpu", typeof(NvHostCtrlGpuDeviceFile) },
+                       { "/dev/nvhost-as-gpu",   typeof(NvHostAsGpuDeviceFile)   },
+                       { "/dev/nvhost-gpu",      typeof(NvHostGpuDeviceFile)     },
+                       //{ "/dev/nvhost-msenc",    typeof(NvHostChannelDeviceFile) },
```

Comment:
```suggestion
                     //{ "/dev/nvhost-msenc",    typeof(NvHostChannelDeviceFile) },
```
And same for other comments.